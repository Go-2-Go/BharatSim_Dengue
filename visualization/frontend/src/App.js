import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { makeStyles } from '@material-ui/core/styles';

import AppRoute from './AppRoute';
import withThemeProvider from './theme/withThemeProvider';

const styles = makeStyles(() => ({
  root: {
    height: '100vh',
    display: 'flex',
    flex: 1,
    flexDirection: 'column',
  },
}));

function App() {
  const classes = styles();
  return (
    <Router>
      <div className={classes.root}>
        <AppRoute />
      </div>
    </Router>
  );
}

export default withThemeProvider(App);
