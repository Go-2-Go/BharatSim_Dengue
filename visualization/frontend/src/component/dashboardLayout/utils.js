import React from 'react';
import renderChart from '../charts/renderChart';

export function renderElement(element) {
  return (
    <div key={element.i} data-grid={element} data-testid={element.i}>
      <div style={{ height: '100%', width: '100%', boxSizing: 'border-box', padding: '10px' }}>
        {renderChart(element.chartType, { config: element.config })}
      </div>
    </div>
  );
}

export function getNewWidgetLayout(numberOfWidgetAdded, cols, count) {
  return createLayout({
    id: `widget-${count}`,
    xPosition: (numberOfWidgetAdded * 2) % cols,
    yPosition: Infinity, // puts it at the bottom
    width: 2,
    height: 2,
  });
}

function createLayout({ id, xPosition, yPosition, width, height }) {
  return {
    i: id,
    x: xPosition,
    y: yPosition,
    w: width,
    h: height,
  };
}
