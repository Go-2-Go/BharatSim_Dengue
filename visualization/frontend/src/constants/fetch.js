export const httpMethods = {
  GET: 'get',
  POST: 'post',
};

export const contentTypes = {
  FILE: 'multipart/form-data',
};
