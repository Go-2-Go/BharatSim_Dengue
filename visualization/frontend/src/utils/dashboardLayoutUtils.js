/* eslint-disable react/prop-types */

import React from 'react';
import renderChart from '../component/charts/renderChart';

export function renderElement({ layout, chartType, config }) {
  return (
    <div key={layout.i} data-grid={layout} data-testid={layout.i}>
      <div style={{ height: '100%', width: '100%', boxSizing: 'border-box', padding: '10px' }}>
        {renderChart(chartType, { config })}
      </div>
    </div>
  );
}

export function getNewWidgetLayout(numberOfWidgetAdded, cols, count) {
  return createLayout({
    id: `widget-${count}`,
    xPosition: (numberOfWidgetAdded * 2) % cols,
    yPosition: Infinity, // puts it at the bottom
    width: 2,
    height: 2,
  });
}

function createLayout({ id, xPosition, yPosition, width, height }) {
  return {
    i: id,
    x: xPosition,
    y: yPosition,
    w: width,
    h: height,
  };
}
