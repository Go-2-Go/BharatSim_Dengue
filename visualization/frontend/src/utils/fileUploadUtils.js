const fileUploadedStatus = {
  LOADING: 'loading',
  SUCCESS: 'success',
  ERROR: 'error',
};

function getStatusAndMessageFor(fileUploadStatus, fileName) {
  return {
    [fileUploadedStatus.ERROR]: {
      status: fileUploadedStatus.ERROR,
      message: `Error occurred while unloading ${fileName}`,
    },
    [fileUploadedStatus.SUCCESS]: {
      status: fileUploadedStatus.SUCCESS,
      message: `${fileName} successfully uploaded`,
    },
    [fileUploadedStatus.LOADING]: {
      status: fileUploadedStatus.LOADING,
      message: `uploading ${fileName}`,
    },
  }[fileUploadStatus];
}

export { fileUploadedStatus, getStatusAndMessageFor };
