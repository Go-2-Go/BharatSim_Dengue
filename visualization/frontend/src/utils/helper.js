function updateState(prevState,updatedValue){
  return Object.assign({},prevState,updatedValue)
}

export {
  updateState,
}