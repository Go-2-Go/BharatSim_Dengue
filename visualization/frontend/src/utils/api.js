import { contentTypes, httpMethods } from '../constants/fetch';
import { url as serviceURL } from './url';
import { fetchData, uploadData } from './fetch';

function headerBuilder({ contentType }) {
  return {
    'content-type': contentType,
  };
}

function formDataBuilder(data) {
  const formData = new FormData();
  data.forEach(({ name, value }) => formData.append(name, value));
  return formData;
}

const api = {
  saveDashboard: async ({ widgets, layout, dashboardId, name, count }) => {
    return uploadData({
      url: serviceURL.DASHBOARD_URL,
      headers: headerBuilder({ contentType: contentTypes.JSON }),
      data: JSON.stringify({
        dashboardData: { widgets, layout, dashboardId, name, count },
      }),
    });
  },

  getAllDashBoard: async () => fetchData({ url: serviceURL.DASHBOARD_URL }),

  uploadFileAndSchema: async ({ file, schema }) =>
    uploadData({
      url: serviceURL.DATA_SOURCES,
      data: formDataBuilder([
        { name: 'datafile', value: file },
        { name: 'schema', value: JSON.stringify(schema) },
      ]),
      headers: headerBuilder({ contentType: contentTypes.FILE }),
    }),

  getCsvHeaders: async (dataSourceId) => fetchData({ url: serviceURL.getHeaderUrl(dataSourceId) }),

  getDatasources: async () => fetchData({ url: serviceURL.DATA_SOURCES }),

  getData: async (datasource, query) =>
    fetchData({ url: serviceURL.getDataUrl(datasource), query }),

  saveProject: async ({ id, ...data }) => {
    const requestObject = {
      url: serviceURL.PROJECT_URL,
      headers: headerBuilder({ contentType: contentTypes.JSON }),
    };
    if (id) {
      return uploadData({
        ...requestObject,
        data: JSON.stringify({ projectData: { ...data, id } }),
        method: httpMethods.PUT,
      });
    }
    return uploadData({
      ...requestObject,
      data: JSON.stringify({ projectData: { ...data } }),
      method: httpMethods.POST,
    });
  },

  getProjects: async () => {
    return fetchData({
      url: serviceURL.PROJECT_URL,
    });
  },
  getProject: async (id) => {
    return fetchData({
      url: serviceURL.getProjectUrl(id),
    });
  },
};

export { api };
