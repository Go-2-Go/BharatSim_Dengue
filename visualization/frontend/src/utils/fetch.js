import axios from 'axios';
import { httpMethods } from '../constants/fetch';

let shouldShowError;

function commonErrorHandler(err) {
  shouldShowError(true);
}

function initApiConfig({ setError }) {
  shouldShowError = setError;
}

function fetchData({ url, method = httpMethods.GET, headers, data, query, customErrorHandler }) {
  return axios({ url, method, headers, data, params: query })
    .then((res) => res.data)
    .catch((err) => {
      if (customErrorHandler) {
        customErrorHandler(err);
        return;
      }
      commonErrorHandler(err);
    });
}

function uploadData({ url, headers, data, query, customErrorHandler }) {
  return fetchData({
    url,
    method: httpMethods.POST,
    headers,
    data,
    params: query,
    customErrorHandler,
  });
}

export { fetchData, uploadData, initApiConfig };
