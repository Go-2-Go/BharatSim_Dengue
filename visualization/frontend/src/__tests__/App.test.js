import React from 'react';
import { act, render } from '@testing-library/react';
import App from '../App';
import { initApiConfig } from '../utils/fetch';

jest.mock('../utils/fetch');

jest.mock('../component/dashboardLayout/DashboardLayout', () => () => <>Dash Board Layout</>);

let mockShouldShowError;

describe('<App />', () => {
  beforeEach(() => {
    initApiConfig.mockImplementation(({ setError }) => {
      mockShouldShowError = setError;
    });
  });
  it('should match snapshot', () => {
    const { container } = render(<App />);

    expect(container).toMatchSnapshot();
  });
  it('should show error with reload button if api is failing', () => {
    const { queryByText } = render(<App />);

    act(() => {
      mockShouldShowError(true);
    });
    const errorComponent = queryByText('Error occurred while loading the page');

    expect(errorComponent).toBeInTheDocument();
  });
  it('should reload the page on click of reload button of error component', () => {
    const reloadFunction = jest.fn();
    const { location } = window;
    delete window.location;
    window.location = { reload: reloadFunction };
    const { getByText } = render(<App />);

    act(() => {
      mockShouldShowError(true);
    });

    const reloadButton = getByText('Reload');
    reloadButton.click();

    expect(reloadFunction).toHaveBeenCalled();
    window.location = location;
  });
});
