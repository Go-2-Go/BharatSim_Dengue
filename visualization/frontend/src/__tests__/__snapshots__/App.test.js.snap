// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div>
      home
    </div>
  </div>
</div>
`;
